### DISCUSSION TITLE
Sealed Secrets with Kustomization

### DISCUSSION THREAD

**Discussion by netthier** [fluxcd/flux2#2875] [Q&A] [answered]
I'd like to deploy a single sealed secret (cluster-wide scope) into multiple namespaces by using multiple Kustomizations.
However, I'm unsure how to deal with the namespaces inside the sealed secret resource.
If my understanding is correct, `metadata.namespace` will be set to whatever `targetNamespace` in the Kustomization is set to, but what about `spec.template.metadata.namespace`? Do I need to configure the Kustomization to patch that, or does it get patched automatically?

**Comment by stefanprodan** [comment, marked as answer]
> Do I need to configure the Kustomization to patch that, or does it get patched automatically?

You need to patch it yourself, Flux has no knowledge about sealed secrets. 

PS. You may want to look into Flux native support for encrypted secrets https://fluxcd.io/docs/guides/mozilla-sops/

**Reply by netthier** [reply]
I'm mirroring a client's infrastructure and they use Sealed Secrets, so SOPS isn't an option for now :/
Thanks for the help!
